### ISSUE TITLE
Reflection no longer works for setting alternative ports.

### ISSUE THREAD

**Issue by athendrix** [TalAloni/SMBLibrary#118]
Your workaround solution in #10 no longer works

Reflection doesn't work for setting static readonly values in .NET Core 3.0 and later.
You get a FieldAccessException

```
System.FieldAccessException: 'Cannot set initonly static field 'DirectTCPPort' after type 'SMBLibrary.Server.SMBServer' is initialized.'
```

This is only because they're both static and readonly.

Of course I could fork the project and do it myself or write a custom SMBServer.cs implementation to add the functionality. But it's really as simple as removing "readonly" from those two values, and 16 characters of difference doesn't seem worth maintaining an entire fork for, and I'd rather use your official NuGet package if possible.

Based on previous closed issues, this clearly isn't a feature you want to add, but there *have* been quite a few of us now who want it for one reason or another, and I don't think it's too crazy.

Is there any way we can convince you? :)

Thanks for all your hard work!

**Comment by TalAloni** [comment]
Hi,
Thanks for letting me know,
so I think I found a solution we can both live with, I've added a private Start overload you can call via reflection and use it to set the port:

```
System.Reflection.MethodInfo startMethod = typeof(SMBLibrary.Server.SMBServer).GetMethod(nameof(SMBLibrary.Server.SMBServer.Start), System.Reflection.BindingFlags.Instance | System.Reflection.BindingFlags.NonPublic);
startMethod.Invoke(m_server, new object[] { serverAddress, transportType, 1445, chkSMB1.Checked, chkSMB2.Checked, false, null });
```
I've already pushed this change and I'll release a new NuGet in a few hours

**Comment by TalAloni** [comment]
I've released v1.4.7 which includes this new overloaded method.

**Comment by tbraun-hk** [comment]
Hi @TalAloni , @athendrix 

after an upgrade to .net6 and the latest version of SMBLibrary, we encountered the same issue for SMB2Client.

A workaround is to again use reflection to call the private "Connect" method which allows to pass the TCP port (in our case 10445 due to the SMB connection being forwarded over SSH where our non-root user is not allowed to use privileged ports).

```
System.Reflection.MethodInfo connectMethod = typeof(SMBLibrary.Client.SMB2Client).
                    GetMethods(System.Reflection.BindingFlags.Instance | System.Reflection.BindingFlags.NonPublic).
                    FirstOrDefault(m => m.Name == "Connect");
                var connected = (bool)connectMethod.Invoke(client, new object[] { new IPAddress(new byte[]{127, 0, 0, 1}), SMBTransportType.DirectTCPTransport, 10445 });
```

It took me quite a while to find this workaround, therefore I wanted to share it with you and everybody else with the need to connect to non-standard SMB ports.

Regards,
Tobias
